app/code/Magento/Review/Model/Review/RatingsProcessor.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Review\Model\Review;
+
+use Magento\Review\Model\ResourceModel\Rating\CollectionFactory as RatingCollectionFactory;
+use Magento\Review\Model\ResourceModel\Rating\Option\CollectionFactory as RatingOptionCollectionFactory;
+
+/**
+ * Class RatingsProcessor
+ *
+ * @package Magento_Review
+ */
+class RatingsProcessor implements RatingsProcessorInterface
+{
+    /**
+     * @var array
+     */
+    private $ratingsByStore;
+
+    /**
+     * @var array
+     */
+    private $ratingOptionsByRating;
+
+    /**
+     * @var RatingCollectionFactory
+     */
+    private $ratingCollectionFactory;
+
+    /**
+     * @var RatingOptionCollectionFactory
+     */
+    private $ratingOptionCollectionFactory;
+
+    /**
+     * RatingsProcessor constructor
+     *
+     * @param RatingCollectionFactory $ratingCollectionFactory
+     * @param RatingOptionCollectionFactory $ratingOptionCollectionFactory
+     */
+    public function __construct(
+        RatingCollectionFactory $ratingCollectionFactory,
+        RatingOptionCollectionFactory $ratingOptionCollectionFactory
+    ) {
+        $this->ratingCollectionFactory = $ratingCollectionFactory;
+        $this->ratingOptionCollectionFactory = $ratingOptionCollectionFactory;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getRatingIdByName(string $ratingName, int $storeId): ?int
+    {
+        if (!isset($this->ratingsByStore[$storeId])) {
+            $collection = $this->ratingCollectionFactory->create();
+            $collection->setStoreFilter($storeId);
+
```

Comment:
We should probably load rating title (name) for specific store ?
If so, we should add
`  $collection->addRatingPerStoreName($storeId);`